contracts/token/SRC20.sol

```diff
@@ -140,17 +148,18 @@ contract SRC20 is ISRC20, ISRC20Owned, ISRC20Managed, SRC20Detailed, Ownable {
      * or address(0) if no rules should be checked on chain.
      * @return True on success.
      */
-    function _updateKYA(bytes32 kyaHash, string memory kyaUrl, address restrictions) internal returns (bool) {
+    function _updateKYA(bytes32 kyaHash, string memory kyaUrl, address restrictions, address rules) internal returns (bool) {
         _kya.kyaHash = kyaHash;
         _kya.kyaUrl = kyaUrl;
 
-        _restrictions = ITransferRules(restrictions);
+        _restrictions = ITransferRestrictions(restrictions);
+        _rules = ITransferRules(rules);
 
-        if (restrictions != address(0)) {
-            require(_restrictions.setSRC(address(this)), "SRC20 contract already set in transfer rules");
+        if (rules != address(0)) {
```

Comment:
I think both rules and restrictions should use same method for dealing with "source" src20 contract address. If there is a need to set it up like this, than we should set up both, the need is the same.